### DISCUSSION TITLE
Ground State Energy Calculation for Weakly Interacting Bose Gas using Variational Monte Carlo

### DISCUSSION THREAD

**Discussion by MrAAafa99** [netket/netket#1688] [Q&A]
Hello,

I have a query regarding Variational Monte Carlo.

I'm working with a weakly interacting Bose gas system and have formulated a trial wavefunction as shown below:

![Screenshot (1278)](https://github.com/netket/netket/assets/54255011/1eb0d943-8074-4b98-a3d9-ab058bdac75e)

In order to fulfill:

![Screenshot (1279)](https://github.com/netket/netket/assets/54255011/91b038cf-b4eb-4bc7-b086-d082ae2ba5fa)

![Screenshot (1280)](https://github.com/netket/netket/assets/54255011/1f2f9749-c3d5-4fe3-bff8-337502af7c99)

 I aim to achieve convergence utilizing Variational Monte Carlo (VMC) within NetKet. Furthermore, I aim to script this trial wavefunction and compute the ground state energy using VMC without employing the pre-existing Jastrow wave function in NetKet. How can I proceed with this process?

Additionally, I'm particularly interested in implementing this for a one-dimensional system. Any guidance on this aspect would be greatly appreciated. Also, could you provide me with the code for this? I'd greatly appreciate having the wave function code so I can utilize it as a model for VMC and obtain the ground state energy. Thank you!

**Comment by PhilipVinc** [comment]
You need to define a NN Module using flax. 
You can find some examples in our tutorial , such as this one https://netket.readthedocs.io/en/latest/tutorials/lattice-fermions.html#slater-determinant

```python
from flax import linen as nn

class MyNet(nn.Module):

    @nn.compact
    def __call__(self, r):
           r_dim = r.shape

           ....
           # return some array with shape r.shape[:-1]
```

**Reply by MrAAafa99** [reply]
Thank you for your explanation. If I understand you correctly, I can use a neural network as a Jastrow factor to multiply it with the Mean field wavefunction and obtain the trial wavefunction. The trial wavefunctions in my mean field section (for example, in one dimension) are Gaussian wavefunctions with a variational coefficient that is to be minimized using variational Monte Carlo. NetKet also provided a snippet of the trial wavefunction for the non-interacting oscillators in the form:
```python
class Gaussian(nn.Module):
    param_dtype: DType = jnp.float64
    kernel_init: NNInitFunc = normal(stddev=1.0)
    
    @nn.compact
    def __call__(self, x_in: Array):
        nv = x_in.shape[-1]

        kernel = self.param("kernel", self.kernel_init, (nv, nv), self.param_dtype)
        kernel = jnp.dot(kernel.T, kernel)

        kernel, x_in = promote_dtype(kernel, x_in, dtype=None)
        y = -0.5 * jnp.einsum("...i,ij,...j", x_in, kernel, x_in)

        return y
```
This wavefunction seems almost identical to my mean-field wavefunction, except here you set the variational coefficient to 0.5. Now, my question is how can I have a trial wavefunction in a way that multiplies a part of the mean field, with the neural network factor (Jastrow factor)?
